pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -713,7 +713,22 @@ void PdBlock_iterBegin(JSOBJ _obj, JSONTypeContext *tc) {
         return;
     }
 
-    arrays = get_sub_attr(obj, "_mgr", "column_arrays");
+    NPY_DATETIMEUNIT dunit = (PyObjectEncoder *)tc->datetimeUnit;
+    PyObject *date_unit;
+    if (dunit == NPY_FR_s) {
+        date_unit = "s";
```

Comment:
What you want to do here is create a Python object from the literal string, so `PyUnicode_FromString("s")`

Reply:
Ignore this comment if you change L730 to `PyObject_CallMethod` - in that case you can just declare `date_unit` as a `char *` and call it a day